src/behavior-considered-undefined.md

```diff
@@ -26,8 +26,15 @@ code.
 * Evaluating a [dereference expression] (`*expr`) on a raw pointer that is
   [dangling] or unaligned, even in [place expression context]
   (e.g. `addr_of!(&*expr)`).
-* Breaking the [pointer aliasing rules]. `&mut T` and `&T` follow LLVM’s scoped
-  [noalias] model, except if the `&T` contains an [`UnsafeCell<U>`].
+* Breaking the [pointer aliasing rules]. `Box<T>`, `&mut T` and `&T` follow LLVM’s
+  scoped [noalias] model, except if the `&T` contains an [`UnsafeCell<U>`].
+  References must not be dangling while they are live. (The exact liveness
+  duration is not specified, but it is certainly upper-bounded by the syntactic
+  lifetime assigned by the borrow checker. When a reference is passed to a
+  function, it is live at least as long as that function call, again except if
```

Comment:
... Does this apply to `Box`? (The intent is *obviously* **no**, but what about for what we're telling LLVM?)

<details><summary>Because it's obviously safe to deallocate a <code>Box</code> allocation within a function,</summary>

```rust
fn example(x: Box<Data>) {
    drop(x);
    // here -- is `x` still considered `dereferencable`?
}
```

</details>

does this mean that `Box` needs to be marked with the hypothetical LLVM-`dereferenceable_on_entry` instead of just LLVM-`dereferenceable`? I suspect this is a nonissue in practice (memory allocation is either a very side-effectful built-in or an opaque environment call preventing code motion over it) but it's definitely *relevant*.

Reply:
See https://github.com/rust-lang/rust/issues/66600 (linked in the PR description) :)